pygfx/utils/transform.py

```diff
@@ -558,61 +549,115 @@ class AffineTransform(AffineBase):
 
     def __init__(
         self,
-        matrix=None,
+        position=(0, 0, 0),
+        rotation=(0, 0, 0, 1),
+        scale=(1, 1, 1),
         /,
         *,
-        position=None,
-        rotation=None,
-        scale=None,
         reference_up=(0, 1, 0),
         is_camera_space=False,
     ) -> None:
         super().__init__(reference_up=reference_up, is_camera_space=is_camera_space)
         self.last_modified = perf_counter_ns()
 
-        if matrix is None:
-            matrix = np.eye(4, dtype=float)
-
-        self.untracked_matrix = np.asarray(matrix)
+        self._position = np.asarray(position, dtype=float)
+        self._rotation = np.asarray(rotation, dtype=float)
+        self._scale = np.asarray(scale, dtype=float)
 
-        if position is not None:
-            self.position = position
-
-        if rotation is not None:
-            self.rotation = rotation
-
-        if scale is not None:
-            self.scale = scale
+        self._position_view = self._position.view()
+        self._position_view.flags.writeable = False
+        self._rotation_view = self._rotation.view()
+        self._rotation_view.flags.writeable = False
+        self._scale_view = self._scale.view()
+        self._scale_view.flags.writeable = False
 
     def flag_update(self):
         self.last_modified = perf_counter_ns()
         super().flag_update()
 
+    @property
+    def position(self) -> np.ndarray:
+        return self._position_view
```

Comment:
Since we re-implement these (`position`, `rotation`, `scane`) here, would it make sense to move their implementation in `AffineBase` (the one doing `return self._decomposed[0]`) to `RecursiveTransform`?

Reply:
I have thought about that and actually went back and forth a few times on that thought. But I think the elegance of the architecture is kept intact this way. The idea of a generalized base transform that can derive anything off a matrix is not valid anymore if I go that route.

Also, it would make the follow-up PR much harder to implement. :) See #920

Reply:
Makes sense!